stripe-core/src/main/java/com/stripe/android/core/networking/DefaultAnalyticsRequestExecutor.kt

```diff
@@ -11,45 +21,137 @@ import kotlin.coroutines.CoroutineContext
 
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
 class DefaultAnalyticsRequestExecutor(
+    private val context: Context,
     private val stripeNetworkClient: StripeNetworkClient,
-    @IOContext private val workContext: CoroutineContext,
-    private val logger: Logger
+    private val workContext: CoroutineContext,
 ) : AnalyticsRequestExecutor {
 
-    constructor() : this(
-        Logger.noop(),
-        Dispatchers.IO
-    )
+    private val canUseWorkManager: Boolean by lazy {
+        val workManagerInClasspath = runCatching {
+            Class.forName("androidx.work.WorkManager")
+        }.isSuccess
+
+        workManagerInClasspath && WorkManager.isInitialized()
```

Comment:
We _could_ initialize `WorkManager` with a default configuration if it’s no initialized yet. Thoughts?

Reply:
There's low chances of the host app depending on it but not initializing it on their side, but we don't lose anything I guess!

Reply:
That's a great approach if we want to use WorkManager outside of analytics